diff --git a/src/time/clock_gettime.c b/src/time/clock_gettime.c
index 4d2ec22f..4e248703 100644
--- a/src/time/clock_gettime.c
+++ b/src/time/clock_gettime.c
@@ -79,7 +79,7 @@ int __clock_gettime(clockid_t clk, struct timespec *ts)
 	r = -ENOSYS;
 	if (sizeof(time_t) > 4)
 		r = __syscall(SYS_clock_gettime64, clk, ts);
-	if (SYS_clock_gettime == SYS_clock_gettime64 || r!=-ENOSYS)
+	if (SYS_clock_gettime == SYS_clock_gettime64 || !r)
 		return __syscall_ret(r);
 	long ts32[2];
 	r = __syscall(SYS_clock_gettime, clk, ts32);
